forms/multilistbox.cpp

```diff
@@ -215,7 +215,8 @@ void MultilistBox::eventOccured(Event *e)
 				{
 					selectedItem = child;
 					selectionAction = selectedSet.find(child) == selectedSet.end();
-					if (selectionAction && funcHandleSelection(e, child, true))
+					bool sel = funcHandleSelection(e, child, true);
+					if (selectionAction && sel)
 					{
```

Comment:
This code break the selection behavior.
Because of drag&drop operations the MouseDown and MouseUp events should handled separately. But the strategy function funcHandleSelection() should called only once per click (MouseDown + MouseUp). In your code the funcHandleSelection() called twice per click, that leads to the broken selection behaviour.

Reply:
No, it does not break. Tested already.

Reply:
The funcHandleSelection() is the strategy function designed by coder. It should called only once per click because it may contain state. In this code the function run twice per click thus the state may switched back and forth. It may matter.